src/functional/fencil_processors/source_modules/cpp_gen.py

```diff
@@ -0,0 +1,102 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2022, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+import ctypes
+from typing import Final, Sequence, Type
+
+import numpy
+
+from functional.fencil_processors import source_modules
+
+
+LANGUAGE_ID: Final = "cpp"
+
+
+def render_python_type(python_type: Type) -> str:
+    mapping: Final = {
+        bool: "bool",
+        int: "long",
+        float: "double",
+        complex: "std::complex<double>",
+        numpy.bool_: "bool",
+        numpy.byte: "signed char",
+        numpy.ubyte: "unsigned char",
+        numpy.short: "short",
+        numpy.ushort: "unsigned short",
+        numpy.intc: "int",
+        numpy.uintc: "unsigned int",
+        numpy.int_: "long",
+        numpy.uint: "unsigned long",
+        numpy.longlong: "long long",
+        numpy.ulonglong: "unsigned long long",
+        numpy.single: "float",
+        numpy.double: "double",
+        numpy.longdouble: "long double",
+        numpy.csingle: "std::complex<float>",
+        numpy.cdouble: "std::complex<double>",
+        numpy.clongdouble: "std::complex<long double>",
+        ctypes.c_bool: "bool",
+        ctypes.c_char: "char",
+        ctypes.c_wchar: "wchar_t",
+        ctypes.c_byte: "char",
+        ctypes.c_ubyte: "unsigned char",
+        ctypes.c_short: "short",
+        ctypes.c_ushort: "unsigned short",
+        ctypes.c_int: "int",
+        ctypes.c_uint: "unsigned int",
+        ctypes.c_long: "long",
+        ctypes.c_ulong: "unsigned long",
+        ctypes.c_longlong: "long long",
+        ctypes.c_ulonglong: "unsigned long long",
+        ctypes.c_size_t: "std::size_t",
+        ctypes.c_ssize_t: "std::ptrdiff_t",
+        ctypes.c_float: "float",
+        ctypes.c_double: "double",
+        ctypes.c_longdouble: "long double",
+    }
+    return mapping[python_type]
+
+
```

Comment:
An optional suggestion in case you didn't know `types.MappingProxyType`:
```suggestion
_PYTHON_TO_CPP_NAME_MAPPING: Final = types.MappingProxyType(
    {
        bool: "bool",
        int: "long",
        float: "double",
        complex: "std::complex<double>",
        numpy.bool_: "bool",
        numpy.byte: "signed char",
        numpy.ubyte: "unsigned char",
        numpy.short: "short",
        numpy.ushort: "unsigned short",
        numpy.intc: "int",
        numpy.uintc: "unsigned int",
        numpy.int_: "long",
        numpy.uint: "unsigned long",
        numpy.longlong: "long long",
        numpy.ulonglong: "unsigned long long",
        numpy.single: "float",
        numpy.double: "double",
        numpy.longdouble: "long double",
        numpy.csingle: "std::complex<float>",
        numpy.cdouble: "std::complex<double>",
        numpy.clongdouble: "std::complex<long double>",
        ctypes.c_bool: "bool",
        ctypes.c_char: "char",
        ctypes.c_wchar: "wchar_t",
        ctypes.c_byte: "char",
        ctypes.c_ubyte: "unsigned char",
        ctypes.c_short: "short",
        ctypes.c_ushort: "unsigned short",
        ctypes.c_int: "int",
        ctypes.c_uint: "unsigned int",
        ctypes.c_long: "long",
        ctypes.c_ulong: "unsigned long",
        ctypes.c_longlong: "long long",
        ctypes.c_ulonglong: "unsigned long long",
        ctypes.c_size_t: "std::size_t",
        ctypes.c_ssize_t: "std::ptrdiff_t",
        ctypes.c_float: "float",
        ctypes.c_double: "double",
        ctypes.c_longdouble: "long double",
    }
)


def render_python_type(python_type: Type) -> str:
    return _PYTHON_TO_CPP_NAME_MAPPING[python_type]
```

Reply:
I made it global in the end.